bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -399,6 +399,42 @@ private static enum State {
         ACTIVE;
     }
 
+    /**
+     * Class for throttling HTTP GET and PUT requests to prevent overloading the Hue bridge.
+     * <p>
+     * The Hue Bridge can get confused if they receive too many HTTP requests in a short period of time (e.g. on start
+     * up), or if too many HTTP sessions are opened at the same time, which cause it to respond with an HTML error page.
+     * So this class a) waits to acquire permitCount (or no more than MAX_CONCURRENT_SESSIONS) stream permits, and b)
+     * throttles the requests to a maximum of one per REQUEST_INTERVAL_MILLISECS.
+     */
+    private class Throttler implements AutoCloseable {
+        private final int permitCount;
+
+        /**
+         * @param permitCount indicates how many stream permits to be acquired.
+         * @throws InterruptedException
+         */
+        Throttler(int permitCount) throws InterruptedException {
+            this.permitCount = permitCount;
+            streamMutex.acquire(permitCount);
+            long delay;
+            synchronized (Clip2Bridge.this) {
+                Instant now = Instant.now();
+                delay = lastRequestTime.map(t -> Duration.between(now, t).toMillis() + REQUEST_INTERVAL_MILLISECS)
+                        .orElse(0L);
+                lastRequestTime = Optional.of(delay > 0 ? now.plusMillis(delay) : now);
```

Comment:
The ternary seems useless, since `now.plusMillis(0)` equals `now`.

Reply:
Nope. Because delay could be negative.

Reply:
OK, I see. I wonder, though, whether it wouldn't be more elegant to add a `.map(d -> Math.max(0L, d))` into line 424 to get rid of both `delay > 0` checks.

Reply:
Hmm. I suppose Thread.sleep and Instant.plusMillisec probably do a zero check internally. So, Ok.

Reply:
Done.